lib/internal/fs/promises.js

```diff
@@ -259,9 +255,6 @@ async function writeFileHandle(filehandle, data, signal) {
   let remaining = data.length;
   if (remaining === 0) return;
   do {
-    if (signal?.aborted) {
```

Comment:
Was there a reason that this was moved? It's supposed to be before the write, as the signal could have been aborted between the time that the file was opened and the time we called writeFileHandle.

Reply:
Maybe I missed something, but changing the write behaviour in a PR that changes the read path seems like something else is wrong. Are you sure that there isn't something else that's wrong? Master is working correctly.

Reply:
Yes, I see what happened. you also accidentally backported the write performance PR

Reply:
I don't think it's terrible that you also backported that, but you need to fix the test. Now that the write limit has increased, the test manages to write everything in "one go" and that's why you had to move the abort check to after the write, because the loop no longer happens.

Reply:
Thank you, I can change it when I'm at home.

Reply:
Note that another option would be to change the `kWriteFileMaxChunkSize` back to `2 ** 14` and add another backport for the other PR, which would maybe make things more orderly.